### PR TITLE
BFD-1923: Add npi org to bfd server project

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-1923](https://jira.cms.gov/browse/BFD-1923)

**User Story or Bug Summary:**

User Story
As a user, I would like organization name to appear in the contained organization resource, so that I can better understand the resource and for CARIN compliance.
 (V2 only) .

 

AC:

Organization Name appears as an attribute in the contained organization resource as looked-up in the NPI coded display file (V2 only)
When not present in the file name should be "Unknown"
Update the DD
Update the API Changelog
 

Implementations Notes:
1. Add the NPI Org nLibrary dependency from AWS Code Artifact.
2. Change the bfd-server project to use that dependency.

---

### What Does This PR Do?

Adds the bfd-data-npi library to the bfd server project

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -101,7 +101,6 @@ I have gone through and verified that...:
     3. Any problems it causes have a small "blast radius".
     4. It'll be easier to rollback if that becomes necessary.
 * [ ] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
-* [ ] The data dictionary has been updated with any field mapping changes, if any were made.
```
Comment (christrotteradhoc) [inline .github/pull_request_template.md:104]: ?

**Review by christrotteradhoc — DISMISSED** [review]

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookup.java [hunk 2]
```diff
@@ -78,7 +78,7 @@ public Optional<String> retrieveNPIOrgDisplay(Optional<String> npiNumber) {
     /*
```
Comment (mjburling) [inline apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookup.java:78]: Merely holding for internal communications. We can resolve this once we're set 😉

**Review by mjburling — APPROVED** [review]

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 3]
```diff
@@ -3478,6 +3623,14 @@ static void addProviderSlice(
       // Certain types have specific systems
       if (type == C4BBOrganizationIdentifierType.NPI) {
         id.setSystem(TransformerConstants.CODING_NPI_US);
+        if (!npiOrgName.isEmpty()) {
+          provider.setName(npiOrgName.get());
+        } else {
+          provider.setName("UNKNOWN");
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3629]: Def thinking this string should be in the constants, since we use it in multiple places

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java [hunk 4]
```diff
@@ -433,6 +434,36 @@ static void addExtensionCoding(
...
+      String codingCode) {
+    IBaseExtension<?, ?> extension = fhirElement.addExtension();
+    extension.setUrl(extensionUrl);
+    if (!codingDisplay.isPresent())
+      extension.setValue(new Coding().setSystem(codingSystem).setCode(codingCode));
+    else
+      extension.setValue(
+          new Coding().setSystem(codingSystem).setCode(codingCode).setDisplay(codingDisplay.get()));
+  }
+
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:466]: Unless we're planning on obsoleting the non-optional version, we can just have this call the other addExtensionCoding method with one line to reduce duplication a little:
```
addExtensionCoding(fhirElement, extensionUrl, codingSystem, codingDisplay.orElse(null), codingCode);
```
Just reducing the chance that a future change in the code doesnt get missed in duplicated locations.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java [hunk 5]
```diff
@@ -759,6 +790,30 @@ static Reference createIdentifierReference(String identifierSystem, String ident
...
+    if (!npiOrgDisplay.isEmpty()) {
+      return new Reference()
+          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue))
+          .setDisplay(npiOrgDisplay.get());
+    } else {
+      return new Reference()
+          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue));
+    }
+  }
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:815]: Since we're using optionals, we could simplify the code a little by using orElse like above:
```
return new Reference()
          .setIdentifier(new Identifier().setSystem(identifierSystem).setValue(identifierValue))
          .setDisplay(npiOrgDisplay.orElse(null));
```
Making some assumptions that display would be null if we didnt set it at all.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import gov.cms.bfd.data.npi.lookup.NPIOrgLookup;
+import java.io.IOException;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+
+/** Integration tests for NPIOrgLookup. */
+public class NPIOrgLookupIT {
+  /** Verifies that it returns a valid npi org */
+  @Test
+  public void VerifyAValidNPIOrg() throws IOException {
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java:14]: We should also have a negative test that checks for UNKNOWN too

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java [hunk 7]
```diff
@@ -78,14 +80,15 @@ public HHAClaim generateClaim() throws FHIRException {
...
+  public void before() throws IOException {
     claim = generateClaim();
     ExplanationOfBenefit genEob =
         HHAClaimTransformerV2.transform(
             new TransformerContext(
                 new MetricRegistry(),
                 Optional.empty(),
-                FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting()),
+                FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting(),
+                NPIOrgLookup.createNpiOrgLookupForTesting()),
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java:91]: Shouldnt each of these transformer tests be testing the new field added here?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java [hunk 8]
```diff
@@ -128,6 +128,7 @@ public void mapEobCommonGroupInpOutHHAHospiceSNFWhenFiNumberExistsExpectExtensio
     TransformerUtilsV2.mapEobCommonGroupInpOutHHAHospiceSNF(
         eob,
         Optional.empty(),
+        Optional.empty(),
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java:131]: We should have a test testing when this is not empty that the correct value is mapped.

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/HHAClaimTransformerTest.java [hunk 9]
```diff
@@ -41,14 +43,15 @@ public void transformSampleARecord() throws FHIRException {
...
-                Optional.empty(),
-                FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting()),
-            claim);
-    assertMatches(claim, eob);
+    TransformerContext transformerContext =
+        new TransformerContext(
+            new MetricRegistry(),
+            Optional.empty(),
+            FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting(),
+            NPIOrgLookup.createNpiOrgLookupForTesting());
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/HHAClaimTransformerTest.java:51]: I thought this change was for v2 only; why are we adjusting the v1 transformer tests?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java [hunk 10]
```diff
@@ -970,6 +970,26 @@ static void assertReferenceIdentifierEquals(
         TransformerUtils.retrieveNpiCodeDisplay(expectedIdentifierValue), reference.getDisplay());
   }
 
+  /**
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java:973]: Missing doc description

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java [hunk 11]
```diff
@@ -970,6 +970,26 @@ static void assertReferenceIdentifierEquals(
...
+      Optional<String> expectedNpiOrgDisplay,
+      Reference reference) {
+    assertTrue(reference.hasIdentifier(), "Bad reference: \" + reference");
+    assertEquals(expectedIdentifierSystem, reference.getIdentifier().getSystem());
+    assertEquals(expectedIdentifierValue, reference.getIdentifier().getValue());
+
+    if (expectedNpiOrgDisplay.isPresent()) {
+      assertEquals(expectedNpiOrgDisplay.get(), reference.getDisplay());
+    }
+  }
```
Comment (lsmitchell) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java:991]: Why are we checking in the v1 transformer that we have npiOrg? (i guess we could have a test checking that it _doesnt_ get mapped, but this isnt doing that I think)

File: apps/bfd-server/dev/api-changelog.md [hunk 12]
```diff
@@ -1,5 +1,69 @@
...
+          "system" : "http://hl7.org/fhir/sid/us-npi",
+          "value" : "0000000000"
+        } ],
+        "active" : true,
+        "name" : "Fake ORG Name"
+      }
+```
+
 ## BFD-2145: Removal of duplicated careteam member entries and their extensions.
 Corrected an issue where duplicate entries for a provider were being supplied in the Explanation of Benefit careTeam element, making it cumbersome to process and link care team members to items.
```
Comment (lsmitchell) [inline apps/bfd-server/dev/api-changelog.md:68]: Why was this deleted?

**Review by lsmitchell — CHANGES_REQUESTED** [review]
I think we're missing some tests for the new code across the touched classes. Also I think v1 things were touched that shouldn't have been (see below)

File: apps/bfd-server/dev/api-changelog.md (hunk 12 above)
Comment (mjburling) [reply apps/bfd-server/dev/api-changelog.md:68]: It's not. I _think_ the json/jsonc codeblock(s) is complaining about the json syntax and isn't closing as a result. The red highlighting, normally depicting removal, seems to be a githubism.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-outpatient.txt [hunk 13]
```diff
@@ -1,2 +1,2 @@
...
-INSERT|567834|1234567890|900|F|W|40|24-JAN-2011|24-JAN-2011|26-FEB-2011|18-FEB-2011|3|999999|1|3|1|15444|A|693.11|11.00|A|KY|1497758544||2222222222||3333333333||4444|0|1|8888.85|6.00|66.89|A40|0|A40|0|A52|0|||||||||||||||||||||||||||||||||||||||||||||||A06|0|A06|0|A15|0|||||||||||||||||||||CD1YYZZ|0|16-JAN-2016|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||A37|0|||||112.00|175.73|693.92|44.00|32490593716374487|373273882012|25|1|03-JAN-1942|CO120|CR121|||0|M99|XX||2||||000000000|111|5|10.45|12.89|15.23|11.00|0|0|200|300|500|5000.00|999.85|134.00| A|77|GG||345345345|
+INSERT|567834|1234567890|900|F|W|40|24-JAN-2011|24-JAN-2011|26-FEB-2011|18-FEB-2011|3|999999|1|3|1|15444|A|693.11|11.00|A|KY|0000000000||2222222222||3333333333||4444|0|1|8888.85|6.00|66.89|A40|0|A40|0|A52|0|||||||||||||||||||||||||||||||||||||||||||||||A06|0|A06|0|A15|0|||||||||||||||||||||CD1YYZZ|0|16-JAN-2016|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||A37|0|||||112.00|175.73|693.92|44.00|32490593716374487|373273882012|25|1|03-JAN-1942|CO120|CR121|||0|M99|XX||2||||000000000|111|5|10.45|12.89|15.23|11.00|0|0|200|300|500|5000.00|999.85|134.00| A|77|GG||345345345|
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-outpatient.txt:2]: yes it is

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/HHAClaimTransformerTest.java (hunk 9 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/HHAClaimTransformerTest.java:51]: Changed for v1 as well since it would be reading from the static file. The mapping stays the same

File: apps/bfd-data-npi/src/test/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookupTest.java [hunk 14]
```diff
@@ -47,4 +47,27 @@ public void shouldNotReturnFakeNPIOrgNameWhenConstructorSetToFalse() throws IOEx
...
+   * Number is empty string.
+   */
+  @Test
+  public void shouldNotReturnWhenNPINumberIEmptyStringAndWhenConstructorSetToTrue()
+      throws IOException {
+    NPIOrgLookup npiOrgDataLookup = NPIOrgLookup.createNpiOrgLookupForTesting();
+    Optional<String> npiOrgDisplay = npiOrgDataLookup.retrieveNPIOrgDisplay(Optional.of(""));
+    assertEquals(false, npiOrgDisplay.isPresent());
+  }
 }
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-data-npi/src/test/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookupTest.java:73]: Line 33 has the positive test case, there is one for production IT in the bfd-server library 
 /** Return Fake NPI Org Name when the parameter bfdServer.include.fake.drug.code is true. */
  @Test
  public void shouldReturnFakeNPIOrgNameWhenConstructorSetToTrue() throws IOException {
    NPIOrgLookup npiOrgDataLookup = NPIOrgLookup.createNpiOrgLookupForTesting();
    Optional<String> npiOrgDisplay =
        npiOrgDataLookup.retrieveNPIOrgDisplay(Optional.of(NPIOrgLookup.FAKE_NPI_NUMBER));
    assertEquals(NPIOrgLookup.FAKE_NPI_ORG_NAME, npiOrgDisplay.get());
  }

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java [hunk 15]
```diff
@@ -101,6 +102,7 @@ private static ExplanationOfBenefit transformClaim(HHAClaim claimGroup) {
         eob,
         C4BBOrganizationIdentifierType.PRN,
         claimGroup.getProviderNumber(),
+        Optional.empty(),
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:105]: Done

[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:105]: done

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java (hunk 11 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java:991]: done

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java (hunk 10 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/stu3/providers/TransformerTestUtils.java:973]: done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java (hunk 15 above)
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:105]: done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 16]
```diff
@@ -298,6 +298,39 @@ static Reference createPractitionerIdentifierReference(
...
+                            .addCoding(
+                                new Coding(type.getSystem(), type.toCode(), type.getDisplay())))
+                    .setValue(value));
+
+    // If this is an NPI perform the extra lookup
+    if (C4BBPractitionerIdentifierType.NPI.equals(type)) {
+      if (npiOrgDisplay.isPresent()) {
+        response.setDisplay(npiOrgDisplay.get());
+      } else {
+        response.setDisplay("UNKNOWN");
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:327]: Done in npiOrgDisplayDefault

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 3 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3629]: Done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 17]
```diff
@@ -3056,7 +3199,8 @@ static void mapEobCommonGroupInpOutHHAHospiceSNF(
                     CcwCodebookVariable.FI_CLM_PROC_DT, procDt)));
 
     // ORG_NPI_NUM => ExplanationOfBenefit.provider
-    addProviderSlice(eob, C4BBOrganizationIdentifierType.NPI, organizationNpi, lastUpdated);
+    addProviderSlice(
+        eob, C4BBOrganizationIdentifierType.NPI, organizationNpi, npiOrgName, lastUpdated);
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:3204]: done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/ExplanationOfBenefitResourceProvider.java [hunk 18]
```diff
@@ -114,6 +116,12 @@ public void setdrugCodeDisplayLookup(FdaDrugCodeDisplayLookup drugCodeDisplayLoo
     this.drugCodeDisplayLookup = drugCodeDisplayLookup;
   }
 
+  /** @param npiOrgLookup the {@link NPIOrgLookup} to use */
+  @Inject
+  public void setnpiOrgLookup(NPIOrgLookup npiOrgLookup) {
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/ExplanationOfBenefitResourceProvider.java:121]: done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java (hunk 4 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:466]: done

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java (hunk 6 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/NPIOrgLookupIT.java:14]: Done it return empty

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java [hunk 19]
```diff
@@ -118,6 +120,7 @@ private static ExplanationOfBenefit transformClaim(HHAClaim claimGroup) {
     TransformerUtilsV2.mapEobCommonGroupInpOutHHAHospiceSNF(
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2.java:120]: don e

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java (hunk 5 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:815]: Need to confirm

Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java:815]: confirmed

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java (hunk 8 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2Test.java:131]: Done

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java (hunk 7 above)
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/HHAClaimTransformerV2Test.java:91]: npi org is now being tested

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java [hunk 20]
```diff
@@ -1988,6 +2021,81 @@ private static CareTeamComponent addCareTeamPractitioner(
...
+   * @param eobItem the {@link ItemComponent} that should be linked to the {@link CareTeamComponent}
+   * @param practitionerIdSystem the {@link Identifier#getSystem()} of the practitioner to reference
+   *     in {@link CareTeamComponent#getProvider()}
+   * @param practitionerIdValue the {@link Identifier#getValue()} of the practitioner to reference
+   *     in {@link CareTeamComponent#getProvider()}
+   * @param careTeamRole the {@link ClaimCareteamrole} to use for the {@link
+   *     CareTeamComponent#getRole()}
+   * @return the {@link CareTeamComponent} that was created/linked
+   */
+  private static CareTeamComponent addCareTeamPractitionerWithNpiOrg(
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java:2041]: Done

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java (hunk 16 above)
[the remark this answers is not in the document]
Comment (lsmitchell) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java]: static strings should be in upper snake case: `NPI_ORG_DISPLAY_DEFAULT`

[the remark this answers is not in the document]
Comment (cbrunefearless) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/TransformerUtilsV2.java]: Changed name to NPI_ORG_DISPLAY_DEFAULT

**Review by lsmitchell — APPROVED** [review]
Thanks for making those changes!

File: apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookup.java (hunk 2 above)
Comment (mjburling) [reply apps/bfd-data-npi/src/main/java/gov/cms/bfd/data/npi/lookup/NPIOrgLookup.java:78]: I think we're all set to deploy this when ready!

**Review by mjburling — APPROVED** [review]
